### ISSUE TITLE
Allow Storing of User Credentials or Tokens Locally to Maintain Persistent Connections

### ISSUE THREAD

**Issue by jsmatias** [aiondemand/aiondemand#13]
Allowing the user credentials or a token (a refresh token or offline token) to be safely stored locally will enable the SDK to maintain a continuous connection to the Metadata Catalogue server. Currently*, users must authenticate regularly to obtain an access token, which expires within 10 minutes. This frequent re-authentication is inconvenient for users and impedes the use of the SDK in background services where there is no user interaction.

## Possible Solutions:

**Store User Credentials Locally**

> Allow user credentials to be stored in a file or environment variable.
> Pros: Easy to implement.
> Cons: Not secure and not ideal for long-term use.

**Store Refresh Token Locally**

> Allow the refresh token to be stored instead of user credentials.
> Pros: More secure than storing user credentials; refresh tokens can be configured for longer lifetimes.
> Cons: Refresh tokens still expire eventually but can have longer validity periods.

**Store Offline Tokens Locally**

> Use offline tokens, which are a type of refresh token that do not expire.
> Pros: Provides long-term access without the need for frequent re-authentication.
> Cons: Requires proper configuration and might require user consent (?) to use offline tokens.

With a refresh token, the idea is to create a client for the SDK on Keycloak and configure it to use a long-lived refresh token. The user only needs to authenticate once to obtain a refresh token, which can then be stored locally. This can be achieved either through a browser or via the command line using the Keycloak API; the latter is my preferred method

---
*The current authentication mechanism refers to the on going PR #12

**Comment by jsmatias** [comment]
Yes, it is configurable. As default, it seems that it is set to 30 days if the client is idle, otherwise it is indefinite.

**Comment by PGijsbers** [comment]
I have seem multiple platforms move away from indefinite tokens, so one with a long life time (maybe selectable by the user) would be best.
